wire-compiler/src/main/java/com/squareup/wire/WireCompiler.java

```diff
@@ -277,7 +309,11 @@ public Void call() throws IOException {
         }
 
         TypeSpec typeSpec = javaGenerator.generateType(type);
-        ClassName javaTypeName = (ClassName) javaGenerator.typeName(type.type());
+        TypeName typeName = javaGenerator.protoFieldsTypeName(type.type());
```

Comment:
Ahh, I get it. It’s because `generateType` does a different thing in the two different cases. Ahh.